### PR TITLE
#351, #352, #357, #358, #361, #363 to prod

### PR DESCRIPTION
**Pull request by jhamman**

### CODE REVIEW

**Review by scottyhq — COMMENTED** [review]
strange. now the google hubs are showing the conda error (https://github.com/pangeo-data/pangeo-cloud-federation/pull/361). I vote to bump repo2docker, put that in staging. Hopefully that will fix this error and we can incorporate into the pull request. What do you think @jhamman ?!

**Comment by jhamman** [comment]
@scottyhq - now testing with r2d master

**Comment by scottyhq** [comment]
@jhamman - try bumping images to latest 2019.08.08 tag (looks like it's a new one b/c its UTC time!)

**Comment by jhamman** [comment]
All green here. @scottyhq, @rsignell-usgs, @rabernat - any objections to merging now?

**Comment by scottyhq** [comment]
actually @jhamman - let's work with the staging images on binder and the hubs for a little bit to make sure these sort of errors are dealt with - https://github.com/pangeo-data/pangeo-cloud-federation/pull/361#issuecomment-519592144 . 

* thanks for the heads up @scollis. which binder / repo are you using?

**Comment by scollis** [comment]
This
https://github.com/openradar/AMS-Open-Source-Radar-2019

I was trying to implement your Tornado version fix when stuff stopped working...

**Comment by scottyhq** [comment]
@jhamman - i'd remove the dask-labextension stuff in the ocean image since it should bring that in from the base images. https://github.com/pangeo-data/pangeo-cloud-federation/blob/93038d7f8631c05f4e020e178666f095d799e50a/deployments/ocean/image/binder/environment.yml#L9

**Comment by scollis** [comment]
@scottyhq Do you expect this PR to fix the conda to docker issue? Or should I be pinging whoever maintains that.. Yet to try on the binder flavoured binder.. I have only tried on the Pangeo binder..

**Comment by scottyhq** [comment]
@scollis - no, this isn't going to affect your binder since you aren't pulling from pangeo-stacks. You could try adding `libiconv` to your https://github.com/openradar/AMS-Open-Source-Radar-2019/blob/master/binder/environment.yml. Or you could re-structure your binder folder to inherit from the pangeo-stacks base image. I prefer this second approach because in theory it guarantees a working configuration of the basic jupyter and dask components. But when we are trying to update things (as we are this week), there are times when this also breaks. For an example see https://github.com/scottyhq/pangeo-binder-test/tree/esip-tech-dive

**Comment by scollis** [comment]
Thanks.. I will add an issue to the pangeo  example notebooks repo.. but you will have the same issue there if you need to rebuild it..

Not this only started happening last night.. it was working fine yesterday

sorry for crashing the PR :)

**Comment by jhamman** [comment]
@scottyhq - https://github.com/pangeo-data/pangeo-cloud-federation/pull/362/commits/6432933930e09a4b6003ce99efce00d68c6e72b9 got staging.hub.pangeo.io running again. The last commit should get the others into the same state.

**Comment by scottyhq** [comment]
great! could you bump staging.nasa.pangeo.io to the same version before merging? https://github.com/pangeo-data/pangeo-cloud-federation/blob/3ac70d70f459c266dc972fc853f453b9d7197a49/deployments/nasa/image/binder/Dockerfile#L2

**Comment by scottyhq** [comment]
@jhamman I think the root of our failing builds is b/c of the conda config. See relevant discussion happening https://github.com/jupyter/repo2docker/issues/731 and https://github.com/jupyter/repo2docker/pull/714. 

I just tried the following test: Launch a functioning base environment and try to run the circleci command manually to update the environment with pangeo-notebook environment.yml. This works without error if you have the current repo2docker conda config:
```
channels:
  - conda-forge
  - defaults
auto_update_conda: false
show_channel_urls: true
update_dependencies: false
channel_priority: flexible
```

But then if you try to install additional packages on top (like we are doing for the hubs) all the incompatibilities of the libraries manifest (just end of log from `conda env update -p /srv/conda/envs/notebook -f pangeo-notebook-environment.yml `):

```
Successfully installed holoviews-1.12.1 testpath-0.3.1 xrviz-0.1.0+83.ge39c38a

#
# To activate this environment, use
#
#     $ conda activate /srv/conda/envs/notebook
#
# To deactivate an active environment, use
#
#     $ conda deactivate

jovyan::dask-tests$ conda info -e
# conda environments:
#
                         /home/jovyan/my-conda-envs/dinosar
base                     /srv/conda
notebook              *  /srv/conda/envs/notebook
```

Add something else or change a version (dask1):
```
jovyan::dask-tests$ conda install dask=1
Collecting package metadata (current_repodata.json): done
Solving environment: failed with current_repodata.json, will retry with next repodata source.
Initial quick solve with frozen env failed.  Unfreezing env and trying again.
Solving environment: failed with current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed
Initial quick solve with frozen env failed.  Unfreezing env and trying again.
Solving environment: \ failed
```

**Comment by scottyhq** [comment]
We're getting evidence of this in the circleci logs:
```
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - conda-forge/linux-64::notebook==6.0.0=py37_0
  - conda-forge/linux-64::ipyleaflet==0.11.1=py37_1
  - conda-forge/noarch::geoviews==1.6.2=py_0
  - conda-forge/linux-64::widgetsnbextension==3.5.1=py37_0
  - conda-forge/noarch::ipywidgets==7.5.1=py_0
  - conda-forge/noarch::jupyter==1.0.0=py_2
  - conda-forge/noarch::nbconvert==5.5.0=py_0
  - conda-forge/noarch::hvplot==0.4.0=py_1
  - conda-forge/noarch::geoviews-core==1.6.2=py_0
  - conda-forge/noarch::nbgitpuller==0.7.0=py_0
  - conda-forge/linux-64::jupyterlab==1.0.4=py37_0
  - conda-forge/noarch::jupyter-server-proxy==1.1.0=py_0
  - conda-forge/noarch::jupyterlab_server==1.0.0=py_1
```

But the ultimate `AssertionError()` isn't very informative

**Comment by jhamman** [comment]
This seems related: https://github.com/ContinuumIO/anaconda-issues/issues/11127 and also has the inconsistent warning.

**Comment by jhamman** [comment]
@scottyhq for the win!

File: deployments/ocean/image/binder/Dockerfile [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # Note that there must be a tag
-FROM pangeo/pangeo-notebook-onbuild:2019.07.01
+FROM pangeo/pangeo-notebook-onbuild:2019.08.08
```
Comment (rabernat) [inline deployments/ocean/image/binder/Dockerfile:2]: This update to ocean, which included numpy and xarray updates, broke several ongoing science projects.

We need to find a way to do these updates in a way that causes less friction for users.

Comment (jhamman) [reply deployments/ocean/image/binder/Dockerfile:2]: @rabernat - you are welcome to rollback this change. Ocean.pangeo.io was having a bunch of problems last week and we made a number of changes (image included) to attempt to fix the problem.

**Pull request merged**